executor/builder.go

```diff
@@ -103,12 +103,83 @@ func (b *executorBuilder) build(p plan.Plan) Executor {
 		return b.buildUnion(v)
 	case *plan.Update:
 		return b.buildUpdate(v)
+	case *plan.Join:
+		return b.buildJoin(v)
 	default:
 		b.err = ErrUnknownPlan.Gen("Unknown Plan %T", p)
 		return nil
 	}
 }
 
+func composeCondition(conditions []ast.ExprNode) ast.ExprNode {
+	length := len(conditions)
+	if length == 0 {
+		return nil
+	} else if length == 1 {
+		return conditions[0]
+	} else {
+		return &ast.BinaryOperationExpr{Op: opcode.EQ, L: composeCondition(conditions[:length/2]), R: composeCondition(conditions[length/2:])}
```

Comment:
For instance, If you construct a + a + ...{10000 items} + a expression(often appears in OLAP query), it will be a left deep tree, and give the pb decoder/encoder a lot of pressure. Currently its advantage is not explicit.